src/util/disposableInstance.ts

```diff
@@ -0,0 +1,18 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+import { Disposable } from 'vscode';
+
+export class DisposableInstance implements Disposable {
```

Comment:
This is just used to force the files to have a method to call so that esbuild doesn't mark the files as unused right? If so, I think it would be a good idea to write that down to document it.

Reply:
Also, consider making `getDisposableInstance()` to avoid the naming clashes. For singleton classes, it's a common pattern that we use in the codebase for the `getInstance()` to get the instanceof the singleton.

Reply:
> This is just used to force the files to have a method to call so that esbuild doesn't mark the files as unused right?

In most cases - yes, that was the purpose of having this class. However there are few classes extending this one instead of implementing `getInstance()` in their body (not needed to be mandatory instantiated at the extension activation)

OK. Will fix.

Reply:
Turned everything back to use `vscode Disposable` and removed `DisposableInstance`

Reply:
> If so, I think it would be a good idea to write that down to document it.

There's a comment describing why we do need these disposables to be created and pushed to subscriptions